ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsTable.tsx

```diff
@@ -1,58 +1,98 @@
+                        <Td dataLabel="Pod ID">{podId}</Td>
+                        <Td dataLabel="Container name">{containerName}</Td>
                     </Tr>
-                )}
-                {deployments.flatMap(({ id, name, namespace, cluster, listeningEndpoints }) =>
-                    listeningEndpoints.map(({ podId, endpoint, signal, containerName }) => (
-                        <Tr key={`${id}/${podId}/${endpoint.port}`}>
-                            <Td dataLabel="Program name">{signal.name}</Td>
-                            <Td dataLabel="PID">{signal.pid}</Td>
-                            <Td dataLabel="Port">{endpoint.port}</Td>
-                            <Td dataLabel="Protocol">{l4ProtocolLabels[endpoint.protocol]}</Td>
+                ))}
+            </Tbody>
+        </TableComposable>
+    );
+}
+
+export type ListeningEndpointsTableProps = {
+    deployments: (ListDeployment & { listeningEndpoints: ProcessListeningOnPort[] })[];
+};
+
+function ListeningEndpointsTable({ deployments }: ListeningEndpointsTableProps) {
+    const expandedRowSet = useSet<string>();
+    return (
+        <TableComposable variant="compact">
+            <Thead noWrap>
+                <Tr>
+                    <Th>{/* Header for expanded column */}</Th>
+                    <Th width={30}>Deployment</Th>
+                    <Th>Namespace</Th>
+                    <Th>Cluster</Th>
+                </Tr>
+            </Thead>
+            {deployments.map(({ id, name, namespace, cluster, listeningEndpoints }, rowIndex) => {
+                const isExpanded = expandedRowSet.has(id);
+                return (
+                    <Tbody key={id} isExpanded={isExpanded}>
+                        <Tr>
+                            {listeningEndpoints.length > 0 ? (
+                                <Td
+                                    expand={{
+                                        rowIndex,
+                                        isExpanded,
+                                        onToggle: () => expandedRowSet.toggle(id),
+                                    }}
+                                />
+                            ) : (
+                                <Td />
+                            )}
                             <Td dataLabel="Deployment">{name}</Td>
                             <Td dataLabel="Namespace">{namespace}</Td>
                             <Td dataLabel="Cluster">{cluster}</Td>
-                            <Td dataLabel="Pod ID">{podId}</Td>
-                            <Td dataLabel="Container name">{containerName}</Td>
                         </Tr>
-                    ))
-                )}
-            </Tbody>
+                        {listeningEndpoints.length > 0 && (
+                            <Tr isExpanded={isExpanded}>
+                                <Td colSpan={4}>
+                                    <Card className="pf-u-m-md" isFlat>
```

Comment:
The `Card` element caught my eye in the `import` statement. For my understanding, did it solve a layout problem? I did not see it in quick look at example code for expandable with nested table:
https://www.patternfly.org/v4/components/table#composable-expandable-with-nested-table

Reply:
This was done to match the mocks:
![image](https://github.com/stackrox/stackrox/assets/1292638/bb6a68b1-3595-4f82-9ee2-4c69dbfae159)

I suppose a utility class for the border would be (or would seem to be) lighter weight, but from what I can tell nothing like that currently exists in PatternFly.

Reply:
We had a discussion with UX, I think it was last year, about card/no-card layouts. 

@mansursyed , @alanonthegit : Could someone on the UX team remind us of the ACS pattern for page layouts?

Reply:
Here is another way to look at it.

**Whenever** we pay an extra development cost in additional elements or props to go beyond what PatternFly gives us for free, we need to explicitly ask and answer the question: Is the benefit in proportion to the cost?

As commented elsewhere in this issue, alignment of columns might be needed anyway.

What is the trade off between investing effort in application-specific alignment that is our responsibility versus enhancing PatternFly?

* PatternFly layout provides consistent indentation and space to indicate a hierarchical level.
* Difference in number and content of columns distinguishes the tables.
* According to gestalt principles of design borders are a speed bump for eyes to enter and exit secondary tables.

Reply:
Removing the Card and using some additional padding would be a very low cost fix that I can push in a follow up. Visually it is very similar, excepting the border and the exact alignment of the left-most columns.
![image](https://github.com/stackrox/stackrox/assets/1292638/f21e077c-8953-4ec4-ac63-68440a8108de)